#152 Removed mysql support in favor of mysqli only

tone2tone opened the pull request:
Searched for occurenced of "cDbDriverMysql" and "mysql_" and changed whenever needed. Deleted old class file. This pull request has not been tested with a separate setup installation.

tone2tone committed:
Removed mysql support in favor of mysqli only

muratpurc commented:
@tone2tone please remove the constant `CON_SETUP_MYSQL` and it's usage in the sources.

tone2tone committed:
#152 Removed mysql support in favor of mysqli only

tone2tone commented:
@purcaholic Thanks for the info - removed CON_SETUP_MYSQL and its usage with a new commit.

muratpurc commented:
@tone2tone  Thank you for the update. While reviewing I've found some occurances of `"mysql"` or `'mysql'` and the function `hasMySQLExtension().` They should be removed, and comments about supported database driver should be adapted to prevent misunderstandings.

But don't remove everything, following files shouldn't be changed:
- examples_misc2.sql
- mysql.js (CodeMirror)

tone2tone commented:
@purcaholic Okay, will review a bit deeper. I came across mysql references that I think have NO mysqli equivalent. Some functions seem to have kept their original names, representing SOME mysql(i) / family reference. hasMySQLExtension is only a function that is never called as it seems. Will delete it.

tone2tone commented:
@purcaholic Before I delete any further, I have questions re occurences that I do not know whether I can safely delete code or not. Here are the things I came across:
./classes/genericdb/class.gdb.mysql.php
does this file hold OLD stuff or actual stuff in an unfortunate named file?

class.item.base.abstract.php, line 20: The pathname should result in something which is no more available, correct? If so, I can delete it:
```
$driver_filename = cRegistry::getBackendPath() . $cfg['path']['classes'] . 'drivers/' . $cfg['sql']['gdb_driver'] . '/class.gdb.' . $cfg['sql']['gdb_driver'] . '.php';
if (cFileHandler::exists($driver_filename)) {
    include_once($driver_filename);
}
```

data/config/production & data/config/test, line 190 both times, loads the following; is it necessary to load something mandatory or can we let go?
`$cfg['sql']['gdb_driver'] = 'mysql';`

muratpurc commented:
@tone2tone You can safely delete the file "class.gdb.mysql.php", since it uses the PHP mysql extension which was deprecated as of PHP 5.5.0 and has been removed as of PHP 7.0.0.

Nobody should set the mysql driver in configuration
`$cfg['sql']['gdb_driver'] = 'mysql';`

This won't work with PHP >= 7 and CONTENIDO should not try to be compatible with PHP 5. 
Removing of the usage of PHP 5 related code is also one of the tasks in current CONTENIDO development.
No regular provider supports PHP 5, end of life of PHP 5.6 was December 2018.

CONTENIDO core code and code of used third party libraries (Smarty, SwiftMailer, etc.) should still work with PHP 5. The question is, should we still keep the old baggage because of possibliy few CONTENIDO installations running on PHP 5, or should we ge rid of it? I prefer to remove old code.. But, this in my personal view on this topic. You can also ask the community, if we should remove PHP 5 support or not.

tone2tone committed:
Removed mysql support in favor of mysqli only (additions)

tone2tone committed:
#152  @tone2tone Removed mysql support in favor of mysqli only (additions02)

tone2tone commented:
@purcaholic I deleted the file plus all mentioned elements from the previous post. In both config.misc.php files, I deleted the whole SECTION for gdb settings, as all other 3-4 variables of that section seem to not be used throughout contenido anymore. 
I very much agree that it makes no sense to keep php5 support and we deleted some < 5.4 support already. These updates now work for 4.10.x family, depending on 7.x, so there's no point IMHO.

tone2tone committed:
Merge branch 'CONTENIDO:develop' into feature/#152-remove-support-for-mysql-extension

muratpurc reviewed (changes requested):
You've removed too much, even files/content which shouln't be removed.

**class.gdb.mysql.php:**
Undo the deltion of this file please, this is not a file which uses PHP mysql extension functions, it contains functions to build join clauses or other MySQL specific operators.

**class.item.base.abstract.php:**
Undo deleted content in this file, it uses the file class.gdb.mysql.php (see above).

**data/config/production/config.autoloader.php and data/config/test/config.autoloader.php:**
Undo the deletion of following line in both files
`'cGenericDbDriverMysql' => 'contenido/classes/genericdb/class.gdb.mysql.php',`

**config.misc.php:**
Undo deleted content in this file, it is used to load the file class.gdb.mysql.php (see above) and countains other important settings.

tone2tone committed:
Removed mysql support in favor of mysqli only

tone2tone committed:
Merge branch 'feature/#152-remove-support-for-mysql-extension' of https://github.com/tone2tone/CONTENIDO into feature/#152-remove-support-for-mysql-extension

tone2tone commented:
@purcaholic Hi Murat, I think that my last commit shows all requested changes including the ones before. Still struggling a bit when I have to do multiple changes multiple times with how to work GIT properly. 
Plz review if this is the commit you can go ahead with ;-)

gonatan commented:
Sorry, I couldn't resist to persist this important point: https://github.com/CONTENIDO/CONTENIDO/wiki

dceonline commented:
checked good work and migrated it to #629 

thx to @tone2tone 

closed
